example/conats.jonprl

```diff
@@ -0,0 +1,53 @@
+Operator iterate : (0; 1; 0).
+[iterate(N; F; B)] =def= [natrec(N; B; _.x.so_apply(F; x))].
+
+Operator top : ().
+[top] =def= [⋂(void; _.void)].
+
+Theorem top-is-top :
+  [⋂(base; x.
+   ⋂(base; y.
+   =(x; y; top)))] {
+  unfold <top>; auto
+}.
+
+Operator conatF : (0).
+[conatF(X)] =def= [+(unit; X)].
+
+Operator conat : ().
+[conat] =def= [⋂(nat; n.iterate(n; x.conatF(x); top))].
+
+Tactic unfolds {
+  *{unfold <conat iterate conatF top>}
+}.
+
+Tactic rauto {
+  *{reduce; auto} ||| May not terminate but shh
+}.
+
+Theorem zero-wf : [∈(inl(<>); conat)] {
```

Comment:
Can we add dedicated operators for `zero` and `succ`? (or whatever names don't clash)